### DISCUSSION TITLE
Running as a service on Pi Zero?

### DISCUSSION THREAD

**Discussion by kapone3047** [Berny23/LD-ToyPad-Emulator#51] [Q&A] [answered]
After much trying and failing with a Pi 400 (as I couldn't get a Pi Zero at the time), I finally have my hands on a Pi Zero W 2 and have got this up and running.

My kids will be the ones using though, so I was hoping to set this up to be literally plug and play. I'm not familiar with node however.

Can someone provide instructions on how to run this as a service? The OS is Raspberry Pi OS Lite

**Comment by Berny23** [comment, marked as answer]
See this reply: https://github.com/Berny23/LD-ToyPad-Emulator/issues/10#issuecomment-950668645

**Comment by VladimirKuletski** [comment]
I use `supervisor`. Just as another option
```
sudo apt install -y supervisor
sudo bash -c 'cat << EOF > /etc/supervisor/conf.d/toypad.conf
[program:toypad]
directory=/home/pi/LD-ToyPad-Emulator
command=/home/pi/.nvm/versions/node/v11.15.0/bin/node index.js
user=pi
autostart=True
autorestart=True
redirect_stderr=True
EOF'
```

**Reply by ignacioat2** [reply]
And after writing this code It will automatically start the node index when i boot up my ps5 connected to my pi.

**Comment by kapone3047** [comment]
Thanks guys

**Comment by ignacioat2** [comment]
Can i ask you how did you got the pi zero w 2 to work. I have a pi zero 2 w and all the code is working but when connecting it to my ps5 it says that cant read toypad. Did yo have this problem? How did you fix it?
I got it luckily running once using some code from a raspberry forum but to be fair i dont know what it does and how to do it without writing code everytime i power my pi. Thanks and i would appreciate your help. This is the website https://forums.raspberrypi.com/viewtopic.php?t=360603#p2163877
